lightningd/pay.c

```diff
@@ -191,6 +191,21 @@ void json_sendpay_fail_fields(struct json_stream *js,
 					fail->msg);
 }
 
+static const char *sendpay_errmsg_fmt(const tal_t *ctx, int pay_errcode,
+				      const struct routing_failure *fail,
+				      const char *details)
+{
+	char *errmsg;
+	if (pay_errcode == PAY_UNPARSEABLE_ONION)
+		errmsg = "Malformed error reply";
+	else {
+		assert(fail);
+		errmsg = tal_fmt(tmpctx, "failed: %s (%s)",
```

Comment:
You passed in a `ctx` to `sendpay_errmsg_fmt`, but you never use it, instead using `tmpctx` here.

Reply:
Thanks @ZmnSCPxj, I've fixed it up, and you can see the change [here](http://46.101.246.115:8888/ElementsProject/lightning/pull/3405/range_diff/dd925d896befd0deceb0bb34fd83b0cb90373fd3..e1264f7f5485ddf21fac52fd7cd6fcc0305e94f2) :+1: